### ISSUE TITLE
Fetch strategy: `highestValue` (to implement most synced)

### ISSUE THREAD

**Issue by dotansimha** [graphprotocol/graph-client#46]
In cases where you might have two sources and you will query both, and only take the most synced (highest block), this might be useful. 

To achieve that, we need to make sure to add `__meta` and block information to outgoing requests: 

```graphql
            _meta {
                block {
                    number
                }
            }
```

 and then check the response based on `_meta.block.number` on both responses? 

```mermaid
graph LR;
    subgraph most-synced
    req(Outgoing Query)-->sA[Subgraph A]; 
    sA-->d{MostSyncedStrategy};
    d-->s1[Source 1]; 
    d-->s2[Source 2]; 
    s1-->synced["process"]
    s2-->synced
    synced-->|"max(_meta.block_number)"|d
    end
```

**Comment by dotansimha** [comment]
Side note: we need to use `chainHeadBlock` from the `graph-node` index GraphQL API

**Comment by wanglonghong** [comment]
This architecture is really awesome. if we could have a health-check endpoint to get `chainHeadBlock`, its also helpful. 
I think the following is an ideal configuration. 

Current configuration. 

>  sources:
>   - name: Connext_Kovan
>     handler:
>       graphql:
>         endpoint: https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan

Proposed configuration
>  sources:
>   - name: Connext_Kovan
>     handler:
>       graphql:
>         queryEndpoint: https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan
>         healthEndpoint: https://api.thegraph.com/index-node/graphql

**Comment by ardatan** [comment]
@wanglonghong I think your feature request is not related to the actual ticket. So please create new tickets for your feature requests and issues instead of writing under an unrelated one. Thanks

**Comment by dotansimha** [comment]
> This architecture is really awesome. if we could have a health-check endpoint to get `chainHeadBlock`, its also helpful.

Yeah, seems like there's a lot to explore around the indexing/admin endpoints, we can use it for a lot more! Thanks!

**Comment by dotansimha** [comment]
Implemented based on `meta._block_number` (in a generic way) in https://github.com/graphprotocol/graph-client/pull/58 .
We can track the support for indexer/admin endpoints here: https://github.com/graphprotocol/graph-client/issues/63

**Comment by dotansimha** [comment]
Released in 0.0.7 :)

**Comment by wanglonghong** [comment]
Nice work!
